### PR TITLE
[SR] Allow RRWeb breadcrumb customization from hybrid SDKs

### PR DESCRIPTION
**Pull request by romtsn**
_#skip-changelog_

## :scroll: Description

* Extract breadcrumb conversion to a separate class, so the hybrid SDKs can override the logic
* Also make the default converter public and open, so others can delegate to it to do the conversion for some categories

### CODE REVIEW

**Review by vaind — APPROVED** [review]
Works fine for RN

**Comment by vaind** [comment]
@romtsn anything blocking this?

**Comment by romtsn** [comment]
> @romtsn anything blocking this?

another reviewer approval and my vacation 😄

File: sentry-android-replay/src/main/java/io/sentry/android/replay/DefaultReplayBreadcrumbConverter.kt [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                timestamp = breadcrumb.timestamp.time
+                breadcrumbTimestamp = breadcrumb.timestamp.time / 1000.0
+                breadcrumbType = "default"
+                category = breadcrumbCategory
+                message = breadcrumbMessage
+                level = breadcrumbLevel
+                data = breadcrumbData
+            }
+        }
+        return rrwebBreadcrumb
```
Comment (markushi) [inline sentry-android-replay/src/main/java/io/sentry/android/replay/DefaultReplayBreadcrumbConverter.kt:111]: looks like rrwebBreadcrumb is always `null` anyway. How about this:
```suggestion
        return if (!breadcrumbCategory.isNullOrEmpty()) {
            RRWebBreadcrumbEvent().apply {
                timestamp = breadcrumb.timestamp.time
                breadcrumbTimestamp = breadcrumb.timestamp.time / 1000.0
                breadcrumbType = "default"
                category = breadcrumbCategory
                message = breadcrumbMessage
                level = breadcrumbLevel
                data = breadcrumbData
            }
        } else {
            null
        }
```

**Review by markushi — APPROVED** [review]
Nice and clean!

**Pull request merged**